demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -2383,14 +2395,33 @@ export class DemoMeetingApp
     return value;
   }
 
-  private async getVoiceFocusDeviceTransformer(): Promise<VoiceFocusDeviceTransformer> {
+  private async getVoiceFocusDeviceTransformer(maxComplexity?: VoiceFocusModelComplexity): Promise<VoiceFocusDeviceTransformer> {
     if (this.voiceFocusTransformer) {
       return this.voiceFocusTransformer;
     }
+
+    function exceeds(configured: VoiceFocusModelComplexity): boolean {
+      const max = Number.parseInt(maxComplexity.substring(1), 10);
+      const complexity = Number.parseInt(configured.substring(1), 10);
+      return complexity > max;
+    }
+
     const logger = new ConsoleLogger('SDK', LogLevel.DEBUG);
-    const transformer = await VoiceFocusDeviceTransformer.create(VOICE_FOCUS_SPEC, { logger });
-    this.voiceFocusTransformer = transformer;
-    return transformer;
+
+    // Find out what it will actually execute, and cap it if needed.
+    const spec: VoiceFocusSpec = { ...VOICE_FOCUS_SPEC };
+    const config = await VoiceFocusDeviceTransformer.configure(spec, { logger });
+
+    let transformer;
+    if (maxComplexity && config.supported && exceeds(config.model.variant)) {
+      logger.info(`Downgrading VF to ${maxComplexity}`);
+      spec.variant = maxComplexity;
+      transformer = VoiceFocusDeviceTransformer.create(spec, { logger });
```

Comment:
Earlier when there was no `maxComplexity?: VoiceFocusModelComplexity` parameter the transformer would be `VoiceFocusDeviceTransformer.create(VOICE_FOCUS_SPEC, { logger });` but now its `VoiceFocusDeviceTransformer.create(spec, { logger }, config);`.

Just checking if this is expected.

Reply:
Yes, it's expected. We're now always making a copy of the spec and potentially changing it, and if we change it we discard the previous `config` in order to do the work again with a fresh spec (e.g., to compute a new model URL).

Reply:
Cool, thanks for explanation 🙏